src/node_report_module.cc

```diff
@@ -0,0 +1,294 @@
+
+  if (options->report_signal == "") options->report_signal = "SIGUSR2";
+}
+
+// A method to sync up data elements in the JS land with its
+// corresponding elements in the C++ world. Required because
+// (i) the tunables are first intercepted through the CLI but
+// later modified via APIs. (ii) the report generation events
+// are controlled partly from C++ and partly from JS.
+void SyncConfig(const FunctionCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  Local<Context> context = env->context();
+  std::shared_ptr<PerIsolateOptions> options = env->isolate_data()->options();
+
+  CHECK_EQ(info.Length(), 2);
+  Local<Object> obj;
+  if (!info[0]->ToObject(context).ToLocal(&obj)) return;
+  bool sync = info[1].As<Boolean>()->Value();
+
+  // Events array
+  Local<String> eventskey = FIXED_ONE_BYTE_STRING(env->isolate(), "events");
+  Local<Value> events_unchecked;
+  if (!obj->Get(context, eventskey).ToLocal(&events_unchecked)) return;
+  Local<Array> events;
+  if (events_unchecked->IsUndefined() || events_unchecked->IsNull()) {
+    events_unchecked = Array::New(env->isolate(), 0);
+    if (obj->Set(context, eventskey, events_unchecked).IsNothing()) return;
+  }
+  events = events_unchecked.As<Array>();
+
+  // Signal
+  Local<String> signalkey = env->signal_string();
+  Local<Value> signal_unchecked;
+  if (!obj->Get(context, signalkey).ToLocal(&signal_unchecked)) return;
+  Local<String> signal;
+  if (signal_unchecked->IsUndefined() || signal_unchecked->IsNull())
+    signal_unchecked = signalkey;
+  signal = signal_unchecked.As<String>();
+
+  Utf8Value signalstr(env->isolate(), signal);
+
+  // Report file
+  Local<String> filekey = FIXED_ONE_BYTE_STRING(env->isolate(), "filename");
+  Local<Value> file_unchecked;
+  if (!obj->Get(context, filekey).ToLocal(&file_unchecked)) return;
+  Local<String> file;
+  if (file_unchecked->IsUndefined() || file_unchecked->IsNull())
+    file_unchecked = filekey;
+  file = file_unchecked.As<String>();
+
+  Utf8Value filestr(env->isolate(), file);
+
+  // Report file path
+  Local<String> pathkey = FIXED_ONE_BYTE_STRING(env->isolate(), "path");
+  Local<Value> path_unchecked;
+  if (!obj->Get(context, pathkey).ToLocal(&path_unchecked)) return;
+  Local<String> path;
+  if (path_unchecked->IsUndefined() || path_unchecked->IsNull())
+    path_unchecked = pathkey;
+  path = path_unchecked.As<String>();
+
+  Utf8Value pathstr(env->isolate(), path);
+
+  // Report verbosity
+  Local<String> verbosekey = FIXED_ONE_BYTE_STRING(env->isolate(), "verbose");
+  Local<Value> verbose_unchecked;
+  if (!obj->Get(context, verbosekey).ToLocal(&verbose_unchecked)) return;
+  Local<Boolean> verbose;
+  if (verbose_unchecked->IsUndefined() || verbose_unchecked->IsNull())
+    verbose_unchecked = Boolean::New(env->isolate(), "verbose");
+  verbose = verbose_unchecked.As<Boolean>();
+
+  bool verb = verbose->BooleanValue(context).FromJust();
```

Comment:
This line prints a deprecation warning.

Reply:
@addaleax - thanks, fixed this; used the `isolate` version.